datatypes.set raises "error: cannot convert 'struct string' to 'int'"

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->

<details>
<summary>V version: V 0.4.11 da0485f, press to see full `v doctor` output</summary>

|V full version      |V 0.4.11 d9afebc277cc3b96c3df0b5b6538cfecffaaa712.da0485f
|:-------------------|:-------------------
|V last modified time|2025-07-24 20:52:15
|                    |
|VTMP                |OK, value: /tmp/v_1000
|Current working dir |OK, value: /tmp
|                    |
|Git version         |git version 2.43.0
|V git status        |weekly.2022.38-6212-gda0485f1-dirty
|.git/config present |true
|                    |
|cc version          |cc (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0
|gcc version         |gcc (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0
|clang version       |N/A
|tcc version         |tcc version 0.9.28rc 2025-02-13 HEAD@f8bd136d (x86_64 Linux)
|tcc git status      |thirdparty-linux-amd64 696c1d84
|emcc version        |N/A
|glibc version       |ldd (Ubuntu GLIBC 2.39-0ubuntu8.5) 2.39

</details>

**What did you do?**
`./v -g -o vdbg cmd/v && ./vdbg  bug.v && /tmp/bug`
```v
import datatypes

fn x() {
	mut set1 := datatypes.Set[string]{}
}

fn y() {
	mut set2 := datatypes.Set[int]{}
}

fn main() {
}

```

**What did you see?**
```
bug.v:4:6: warning: unused variable: `set1`
    2 | 
    3 | fn x() {
    4 |     mut set1 := datatypes.Set[string]{}
      |         ~~~~
    5 | }
    6 |
bug.v:8:6: warning: unused variable: `set2`
    6 | 
    7 | fn y() {
    8 |     mut set2 := datatypes.Set[int]{}
      |         ~~~~
    9 | }
   10 |
================== C compilation error (from tcc): ==============
cc: /tmp/v_1000/bug.01K15GY6GXKE8D5ZTSB1A55K62.tmp.c:6636: error: cannot convert 'struct string' to 'int'
```

**What did you expect to see?**

bug.v compiles successfully

--- a/vlib/v/gen/c/infix.v
+++ b/vlib/v/gen/c/infix.v
@@ -614,7 +614,15 @@ fn (mut g Gen) infix_expr_in_op(node ast.InfixExpr) {
 	} else if right.unaliased_sym.kind == .map {
 		g.write('_IN_MAP(')
 		if !left.typ.is_ptr() {
-			styp := g.styp(node.left_type)
+			mut sym_map := g.table.sym(node.right_type)
+			if sym_map.info is ast.Alias {
+				sym_map = g.table.sym((sym_map.info as ast.Alias).parent_type)
+			}
+			styp := g.styp(if sym_map.info is ast.Map {
+				(sym_map.info as ast.Map).key_type
+			} else {
+				node.left_type
+			})
 			g.write('ADDR(${styp}, ')
 			g.expr(node.left)
 			g.write(')')

--- a/vlib/v/tests/builtin_maps/map_generic_keys_with_in_op_test.v
+++ b/vlib/v/tests/builtin_maps/map_generic_keys_with_in_op_test.v
@@ -1,0 +1,18 @@
+import datatypes
+
+fn x() {
+	mut set1 := datatypes.Set[string]{}
+	set1.add('')
+	assert set1.exists('')
+}
+
+fn y() {
+	mut set2 := datatypes.Set[int]{}
+	set2.add(1)
+	assert set2.exists(1)
+}
+
+fn test_main() {
+	x()
+	y()
+}

